UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs

```diff
@@ -912,6 +912,13 @@ public static object DeSerialiseValue(object InObject, string StringVariable, Ty
 				}
 				else
 				{
+					if (InType == null || InObject == null || InObject as IConvertible == null)
+					{
+						Logger.LogError($"Can't convert {StringVariable} to {InObject.GetType()}  " +
+							$"[(InType == null) = {InType == null} || (InObject == null) == {InObject == null} || (InObject as IConvertible == null) = {InObject as IConvertible == null}]", Category.NetUI);
+						return null;
+					}
+
 					return Convert.ChangeType(StringVariable, InObject.GetType());
```

Comment:
InvalidCastException: Invalid cast from 'System.String' to 'DamageType'.
System.Convert.DefaultToType (System.IConvertible value, System.Type targetType, System.IFormatProvider provider) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.String.System.IConvertible.ToType (System.Type type, System.IFormatProvider provider) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Convert.ChangeType (System.Object value, System.Type conversionType, System.IFormatProvider provider) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.Convert.ChangeType (System.Object value, System.Type conversionType) (at <437ba245d8404784b9fbab9b439ac908>:0)
Librarian+Page.DeSerialiseValue (System.Object InObject, System.String StringVariable, System.Type InType) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:915)
Librarian+Page.SetValue (System.String Value) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:889)
Librarian.PageSetValue (Librarian+Page Page, System.String newValue) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:303)
VariableViewer.RequestChangeVariable (System.UInt64 PageID, System.String ChangeTo, System.Boolean SendToClient, UnityEngine.GameObject WhoBy, System.String AdminId) (at /github/workspace/UnityProject/Assets/Scripts/Variable Viewer/BookViewer/VariableViewer.cs:233)
RequestChangeVariableNetMessage.ValidateAdmin () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/RequestChangeVariableNetMessage.cs:24)
RequestChangeVariableNetMessage.Process () (at /github/workspace/UnityProject/Assets/Scripts/Messages/Client/VariableViewer/RequestChangeVariableNetMessage.cs:17)
GameMessageBase.Process (Mirror.NetworkConnection sentBy) (at /github/workspace/UnityProject/Assets/Scripts/Messages/GameMessageBase.cs:25)
ClientMessage.Process (Mirror.NetworkConnection sentBy) (